services/QuillLMS/app/controllers/diagnostic_insights_controller.rb

```diff
@@ -17,6 +19,22 @@ def diagnostics_assigned
     render json: fetch_diagnostics_assigned.to_json(root: false)
   end
 
+  def complete_history
+    # TODO: make this cache smarter or eradicate the need for it by speeding up the query
+    complete_history_by_classroom_data = Rails.cache.fetch("#{COMPLETE_HISTORY_CACHE_KEY}.#{current_user.id}", expires_in: 24.hours) do
+      complete_history_by_classroom = {}
+      current_user.classrooms_i_teach.each do |classroom|
```

Comment:
You can use `each_with_object`:

 `current_user.classrooms_i_teach.each_with_object({}) do |classroom, complete_history_by_classroom|`

which will allow for removal of the outer temp variable.